app/views/admin/financial_reports/_form_fields.html.erb

```diff
@@ -0,0 +1,53 @@
+<%= render "govuk_publishing_components/components/input", {
+  label: {
+    text: "Year (required)"
+  },
+  name: "financial_report[year]",
+  id: "financial_report_year",
+  value: form.object.year,
+  width: 4,
+  heading_level: 2,
+  heading_size: "l",
+  error_items: errors_for(form.object.errors, :year)
+} %>
+
+<%= render "govuk_publishing_components/components/input", {
+  label: {
+    text: "Funding"
+  },
+  name: "financial_report[funding]",
+  id: "financial_report_funding",
+  value: form.object.funding,
```

Comment:
As discussed this will fix the weird behaviour with their input turning to 0 when the validation calls `.to_i` on the string
```
value: params.dig("financial_report", "year") || form.object.year,
```

Reply:
Great, that's a lot better. Cheers! 👍